File: frontend/lib/models/message_model.dart
class MessageModel {
  final int? id;
  final int? sender;
  final String? receiver;
  final bool istext;
  final String? timestamp;
  final List<int> content; 
  final String? sendername;
  final bool? isOffline;
  final String uniqueid;
  final String? replyto;

  MessageModel({
    this.id,
    this.sender,
    this.receiver,
    required this.istext,
    this.timestamp,
    required this.content,
    this.sendername,
    this.isOffline,
    this.replyto,
    required this.uniqueid
  });
}

File: backend/model/message_model.go
package model

type MessageModel struct {
	ID         int    `json:"id"`
	Content    []byte `json:"content"`
	Sender     int    `json:"sender"`
	Receiver   string `json:"receiver"`
	IsText     bool   `json:"istext"`
	Timestamp  string `json:"timestamp"`
	SenderName string `json:"sendername"`
}
